### PR TITLE
[#164165113] Update react-navigation to v3

### PR DESCRIPTION
**Pull request by fpersico**
App works like before (custom navigationHistory included)

### CODE REVIEW

File: ts/boot/configureStoreAndPersistor.ts [hunk 1]
```diff
@@ -92,11 +92,11 @@ const sagaMiddleware = createSagaMiddleware();
  * @https://github.com/react-navigation/react-navigation/issues/3438.
  */
 const navigation = createReactNavigationReduxMiddleware(
+  // This is a selector to get the navigation state from the global state
```
Comment (cloudify) [inline ts/boot/configureStoreAndPersistor.ts]: is this change necessary? looks like just a reordering of the rows

File: ts/navigation/MainNavigator.tsx [hunk 2]
```diff
@@ -139,30 +139,36 @@ const navigation = createBottomTabNavigator(
...
+                  options.tintColor === null ? undefined : options.tintColor
+              }
             ]}
           >
             {getLabel(routeName)}
           </Text>
         );
       },
-      tabBarIcon: ({ tintColor }) => {
+      tabBarIcon: (options: { tintColor: string | null; focused: boolean }) => {
```
Comment (cloudify) [inline ts/navigation/MainNavigator.tsx]: TS doesn't infer the type of the function here?

File: ts/boot/configureStoreAndPersistor.ts (hunk 1 above)
Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: It is a reordering of the params
oldversion -> key, selector
newversion -> selector, key

File: ts/navigation/MainNavigator.tsx (hunk 2 above)
Comment (fpersico) [reply ts/navigation/MainNavigator.tsx]: Typescript infer "any" so i have to specify the exact type

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#164165113](https://www.pivotaltracker.com/story/show/164165113): Aggiornare react-navigation alla versione 3


New dependencies added: <a href='https://www.npmjs.com/package/react-native-gesture-handler'><code>react-native-gesture-handler</code></a>.


<h2><a href="https://github.com/kmagiera/react-native-gesture-handler#readme">react-native-gesture-handler</a></h2>
<p>Author: Krzysztof Magiera</p>
<p>Description: Experimental implementation of a new declarative API for gesture handling in react-native</p>
<p>Homepage: <a href="https://github.com/kmagiera/react-native-gesture-handler#readme">https://github.com/kmagiera/react-native-gesture-handler#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 2 years ago</td></tr><tr><td>Last Updated</td><td>1 day ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>62</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/hoist-non-react-statics'><code>hoist-non-react-statics</code></a>, <a href='https://www.npmjs.com/package/invariant'><code>invariant</code></a> and <a href='https://www.npmjs.com/package/prop-types'><code>prop-types</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

<p align="center">
  
  <h1 align="center">React Native Gesture Handler</h1>
  <h3 align="center">Declarative API exposing platform native touch and gesture system to React Native.</h3>
</p>

React Native Gesture Handler provides native-driven gesture management APIs for building best possible touch-based experiences in React Native.

With this library gestures are no longer controlled by the JS responder system, but instead are recognized and tracked in the UI thread.
It makes touch interactions and gesture tracking not only smooth, but also dependable and deterministic.

## Installation

Check [getting started](https://kmagiera.github.io/react-native-gesture-handler/docs/getting-started.html) section of our docs for the detailed installation instructions.

## Documentation

Check out our dedicated documentation page for info about this library, API reference and more: [https://kmagiera.github.io/react-native-gesture-handler](https://kmagiera.github.io/react-native-gesture-handler)

## Examples

If you want to play with the API but don't feel like trying it on a real app, you can run the example project. Clone the repo, go to the `Example/` folder and run:
```bash
  yarn install
```

Then run `react-native run-android` or `react-native run-ios` (depending on which platform you want to run the example app on).

You will need to have an Android or iOS device or emulator connected as well as `react-native-cli` package installed globally.

## License

Gesture handler library is licensed under [The MIT License](LICENSE).

## Credits

This project is supported by amazing people from [Expo.io](https://expo.io) and [Software Mansion](https://swmansion.com)

[expo](https://expo.io)
[swm](https://swmansion.com)


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>
